core/src/main/java/nu/yona/server/subscriptions/service/BuddyDto.java

```diff
@@ -40,51 +38,37 @@
 	private final UUID id;
 	private final UserDto user;
 	private final String personalInvitationMessage;
-	private final String nickname;
 	private final Optional<UUID> userAnonymizedId;
 	private final Optional<LocalDate> lastMonitoredActivityDate;
 	private final Status sendingStatus;
 	private final Status receivingStatus;
-	private final Optional<Set<GoalDto>> goals;
 	private final LocalDateTime lastStatusChangeTime;
 
-	public BuddyDto(UUID id, UserDto user, String nickname, Optional<UUID> userAnonymizedId,
-			Optional<LocalDate> lastMonitoredActivityDate, Status sendingStatus, Status receivingStatus,
-			LocalDateTime lastStatusChangeTime)
+	public BuddyDto(UUID id, UserDto user, Optional<UUID> userAnonymizedId, Optional<LocalDate> lastMonitoredActivityDate,
+			Status sendingStatus, Status receivingStatus, LocalDateTime lastStatusChangeTime)
 	{
-		this(id, user, null, nickname, userAnonymizedId, lastMonitoredActivityDate, sendingStatus, receivingStatus,
-				lastStatusChangeTime, Optional.empty());
+		this(id, user, null, userAnonymizedId, lastMonitoredActivityDate, sendingStatus, receivingStatus, lastStatusChangeTime);
 	}
 
 	public BuddyDto(UserDto user, String personalInvitationMessage, Status sendingStatus, Status receivingStatus,
 			LocalDateTime lastStatusChangeTime)
 	{
-		this(null, user, personalInvitationMessage, null, Optional.empty(), Optional.empty(), sendingStatus, receivingStatus,
-				lastStatusChangeTime, Optional.empty());
-	}
-
-	private BuddyDto(UUID id, UserDto user, String nickname, Optional<UUID> userAnonymizedId,
-			Optional<LocalDate> lastMonitoredActivityDate, Status sendingStatus, Status receivingStatus,
-			LocalDateTime lastStatusChangeTime, Set<GoalDto> goals)
-	{
-		this(id, user, null, nickname, userAnonymizedId, lastMonitoredActivityDate, sendingStatus, receivingStatus,
-				lastStatusChangeTime, Optional.of(goals));
+		this(null, user, personalInvitationMessage, Optional.empty(), Optional.empty(), sendingStatus, receivingStatus,
+				lastStatusChangeTime);
 	}
 
-	private BuddyDto(UUID id, UserDto user, String personalInvitationMessage, String nickname, Optional<UUID> userAnonymizedId,
+	private BuddyDto(UUID id, UserDto user, String personalInvitationMessage, Optional<UUID> userAnonymizedId,
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Constructor has 8 parameters, which is greater than 7 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://10.127.10.107:9000/coding_rules#rule_key=squid%3AS00107)